### PR TITLE
update most packages I maintain

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

nix-build -A openmpt123 -A vkquake -A franz -A openmw-tes3mp -A kmsxx -A openjk -A mytetra -A qdirstat -A python27Packages.kmsxx -A python37Packages.kmsxx -A ps_mem -A vk-messenger

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
openmpt123 still fails to build: https://gist.github.com/Mic92/0345c5294f04134388177780648e16d5

**Comment by gnidorah** [comment]
@Mic92 do we have issues with static libraries in NixOS? I see `--disable-static` is passed by default. Switched to shared, should build now.

**Comment by Mic92** [comment]
@gnidorah we don't build static libraries usually like most linux distributions, but they should work. There is a static overlay called pkgsStatic that enables static linking by default in some packages.

File: pkgs/applications/audio/openmpt123/default.nix [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
-  buildInputs = [ SDL2 flac libsndfile ];
-  makeFlags = [ "NO_PULSEAUDIO=1 NO_LTDL=1 TEST=0 EXAMPLES=0" ]
-  ++ stdenv.lib.optional (stdenv.isDarwin) "SHARED_SONAME=0";
-  installFlags = "PREFIX=\${out}";
+
+  enableParallelBuilding = true;
+  doCheck = true;
+
+  buildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ zlib mpg123 libogg libvorbis portaudio libsndfile flac ];
```
Comment (FRidh) [inline pkgs/applications/audio/openmpt123/default.nix]: `buildInputs` and `nativeBuildInputs` are the other way around

File: pkgs/applications/audio/openmpt123/default.nix [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
+  doCheck = true;
+
   nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ SDL2 flac libsndfile ];
-  makeFlags = [ "NO_PULSEAUDIO=1 NO_LTDL=1 TEST=0 EXAMPLES=0" ]
-  ++ stdenv.lib.optional (stdenv.isDarwin) "SHARED_SONAME=0";
-  installFlags = "PREFIX=\${out}";
+  buildInputs = [ zlib mpg123 libogg libvorbis portaudio libsndfile flac ];
+
+  configureFlags = [
```
Comment (FRidh) [inline pkgs/applications/audio/openmpt123/default.nix]: Note we often adjust this flag using `usePulseAudio = config.pulseaudio or stdenv.isLinux;`

Comment (gnidorah) [reply pkgs/applications/audio/openmpt123/default.nix]: @FRidh Done, thanks!

**Comment by gnidorah** [comment]
Merge please?

**Comment by gnidorah** [comment]
@FRidh @Mic92 Thanks!

**Pull request merged**